R/gee-tidiers.R

```diff
@@ -0,0 +1,135 @@
+#' @templateVar class gee
+#' @template title_desc_tidy
+#'
+#' @param x A `gee` object returned from a call to [gee::gee()].
+#' @param conf.int 	Logical indicating whether or not to include a confidence
+#'   interval in the tidied output. Defaults to FALSE.
+#' @param conf.level The confidence level to use for the confidence interval if
+#'   conf.int = TRUE. Must be strictly greater than 0 and less than 1. Defaults
+#'   to 0.95, which corresponds to a 95 percent confidence interval.
+#' @param exponentiate Logical indicating whether or not to exponentiate the the
+#'   coefficient estimates. This is typical for logistic and multinomial
+#'   regressions, but a bad idea if there is no log or logit link. Defaults to
+#'   FALSE.
+#' @param ... Additional arguments. Not used. Needed to match generic signature
+#'   only. Cautionary note: Misspelled arguments will be absorbed in ..., where
+#'   they will be ignored. If the misspelled argument has a default value, the
+#'   default value will be used. For example, if you pass conf.level = 0.9, all
+#'   computation will proceed using conf.level = 0.95. Additionally, if you pass
+#'   newdata = my_tibble to an augment() method that does not accept a newdata
+#'   argument, it will use the default value for the data argument.
+#' @details If `conf.int = TRUE`, the confidence interval is computed with an
+#'   internal `confint.gee()` function.
+#'
+#'   If you have missing values in your model data, you may need to refit the
+#'   model with `na.action = na.exclude` or deal with the missingness in the
+#'   data beforehand.
+#'   
+#' @examples
+#'
+#' library(gee)
+#' 
+#' df <- mtcars
+#'
+#' gee_mod <- gee(
+#'   mpg ~ hp + wt, id = cyl, data = df, 
+#'   family = gaussian, corstr = "independence"
+#' )
+#' 
+#' tidy(gee_mod)
+#' tidy(gee_mod, conf.int = TRUE)
+#' glance(gee_mod)
+#' augment(gee_mod)
+#'
+#' @evalRd return_tidy(regression = TRUE)
+#'
+#' @export
+#' @aliases gee_tidiers 
+#' @seealso [tidy()], [gee::gee()]
+tidy.gee <- function(x, conf.int = FALSE, conf.level = .95, exponentiate = FALSE, ...) {
+  
+  coefs <-
+    stats::coef(summary(x)) %>%
+    tibble::as_tibble(rownames = "term") %>%
+    dplyr::select(-c(`Naive S.E.`, `Naive z`)) 
+  
+  # Remove naive estimators
+  names(coefs) <- c("term", "estimate", "std.error", "statistic")
+  
+  if (conf.int) {
+    zstat <- qnorm((1 + conf.level)/2)
+    coefs$conf.low <- coefs$estimate - zstat * coefs$std.error
+    coefs$conf.high <- coefs$estimate + zstat * coefs$std.error
+  }
+  
+  if (exponentiate) {
```

Comment:
We have a new approach to coefficient exponentiation which it would be nice to use here. See https://github.com/tidymodels/broom/blob/master/R/biglm-tidiers.R#L57 for an example.